scripts/terrain.gd

```diff
@@ -38,8 +43,14 @@ func generate_terrain():
 	surface_tool.create_from(array_mesh,0)
 	surface_tool.generate_normals()
 	
-	$MeshInstance3D.mesh = surface_tool.commit()
-	$CollisionShape3D.shape = array_mesh.create_trimesh_shape()
+	$Terrain/MeshInstance3D.mesh = surface_tool.commit()
+	$Terrain/CollisionShape3D.shape = array_mesh.create_trimesh_shape()
+	
+	# Give the $DitzyTerrain/CollisionShape3D time to instantiate the collision mesh before placing buildings
+	call_deferred("_emit_landscape_complete")
```

Comment:
[minor]
An alternative syntax for this is `_emit_landscape_complete.call_deferred()` which is preferable because if you ever change the name of the function but forget to update the call it will give you an error there-and-then, instead of generating a runtime error later. It's quicker and easier to debug if you know about the error sooner while the context is still fresh in your head.